arch/src/x86_64/mod.rs

```diff
@@ -39,6 +40,19 @@ pub enum Error {
     ZeroPageSetup,
 }
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            Error::E820Configuration => write!(f, "Invalid e820 setup params."),
+            Error::MpTableSetup(err) => write!(f, "Error writing MP table to memory: {}", err),
+            Error::ZeroPagePastRamEnd => {
+                write!(f, "The zero page extends past the end of guest_mem.")
+            }
+            Error::ZeroPageSetup => write!(f, "Error writing the zero page of guest memory."),
```

Comment:
It Should be `Error writing the zero page to guest memory.` The comment from aboce the correspondent enum item should be updated too.